### DISCUSSION TITLE
How can I add custom "tags" to mdast-util-from-markdown?

### DISCUSSION THREAD

**Discussion by alvarocastro** [syntax-tree/unist#135] [Q&A]
I need to parse markdown, but my markdown has some custom tags (or nodes? I don't know the proper name) which are `[%12345%]` for mentions (where the number is the user ID) and `#text` (for hashtags).

I would like to know if `mdast-util-from-markdown` can somehow be extended to add those custom tags and get them as custom defined nodes in the result. The same way `**text**` ends up being a `bold` node.

---

I'm open to use other libraries, I just need to parse markdown (with some custom tags) into some kind of AST to conveniently manipulate it later (format it into some other kind of AST for a WYSIWYG editor (TipTap / ProseMirror))

**Comment by ChristianMurphy** [comment]
Welcome @alvarocastro! 👋 
Yes additional extensions are possible, but not recommended, reasons: https://github.com/micromark/micromark?tab=readme-ov-file#extending-markdown
If you do choose the path of creating yet another markdown flavor and training authors, you can create a micromark extension https://github.com/micromark/micromark?tab=readme-ov-file#creating-a-micromark-extension then transform those tokens into an AST

You can find an example of this in https://github.com/syntax-tree/mdast-util-math/#readme, which extends markdown with math syntax.

**Comment by wooorm** [comment]
You might also be able to add them with https://github.com/syntax-tree/mdast-util-find-and-replace (or https://github.com/syntax-tree/hast-util-find-and-replace, if you care about HTML)
